trunk/SUAVE/Components/Energy/Converters/Propeller.py

```diff
@@ -314,22 +696,100 @@ def spin(self,conditions):
         conditions.propulsion.etap = etap
         
         # store data
-        results_conditions = Data      
-        conditions.propulsion.acoustic_outputs = results_conditions(
-            number_sections    = N,
-            r0                 = r,
-            airfoil_chord      = c,
-            blades_number      = B,
-            propeller_diameter = D,
-            drag_coefficient   = Cd,
-            lift_coefficient   = Cl,
-            omega              = omega,
-            velocity           = V,
-            thrust             = thrust,
-            power              = power,
-            mid_chord_aligment = self.prop_attributes.mid_chord_aligment
-        )
+        results_conditions = Data     
+        outputs   = results_conditions(
+            n_blades                  = B,
+            R                         = R,
+            D                         = D,
+            number_sections           = N,
+            radius_distribution       = np.linspace(Rh ,R, N),
+            chord_distribution        = c,     
+            twist_distribution        = beta,            
+            r0                        = r,
+            thrust_angle              = theta,
+            speed_of_sound            = conditions.freestream.speed_of_sound,
+            density                   = conditions.freestream.density,
+            velocity                  = Vv, 
+            vt                        = vt, 
+            va                        = va, 
+            drag_coefficient          = Cd,
+            lift_coefficient          = Cl,       
+            omega                     = omega,          
+            
+            blade_dT_dR               = rho*(Gamma*(Wt-epsilon*Wa)),   
+            blade_dT_dr               = rho*(Gamma*(Wt-epsilon*Wa))*R,  
+            blade_T_distribution      = rho*(Gamma*(Wt-epsilon*Wa))*deltar, 
+            blade_T                   = thrust/B,  
+            Ct                        = 0, 
+            Cts                       = 0, 
+            
+            blade_dQ_dR               = rho*(Gamma*(Wa+epsilon*Wt)*r), 
+            blade_dQ_dr               = rho*(Gamma*(Wa+epsilon*Wt)*r)*R,
+            blade_Q_distribution      = rho*(Gamma*(Wa+epsilon*Wt)*r)*deltar,
+            blade_Q                   = torque/B,   
+            Cq                        = 0, 
+            
+            power                     = power,
+            
+            mid_chord_aligment        = self.mid_chord_aligment     
+        ) 
+        
+        return thrust, torque, power, Cp, outputs  , etap  
+
+    
+    def spin_surrogate(self,conditions):
         
```

Comment:
Needs a docstring, especially since the function name is unclear.

Reply:
Header added